### PR TITLE
add date-time option for easy testing conditional-turn-restriction

### PR DESCRIPTION
**Pull request by nvcnvn**
## 👨‍💻 Changes proposed
- Add date-time picker (for easier testing time-based turn restriction)

## 📷 Screenshots
Screenshot 2023-10-25 at 15 37 01

### CODE REVIEW

**Comment by nilsnolde** [comment]
Thanks, that'd be a nice addition which would make the app much more useful. A couple of comments:
- seems like it's now mandatory to give a date/time? Would be better if it was opt-in, e.g. the default item in the dropdown could be "Anytime" or so and a greyed out date/time field. WDYT?
- the `moment` library causes a build failure: https://github.com/gis-ops/valhalla-app/actions/runs/6637880358/job/18036721164?pr=200. Can you have a look?

**Comment by nvcnvn** [comment]
> * seems like it's now mandatory to give a date/time? Would be better if it was opt-in, e.g. the default item in the dropdown could be "Anytime" or so and a greyed out date/time field. WDYT?

I just want to clarify the behavior of Valhalla in this case, should "Anytime" (empty date_time)  different with "Current" (date_time.type = 0)?
Because most of the time we: A don't care about time, B want to test departure/arrival.

**Comment by nilsnolde** [comment]
Yes it will be different. If you say datetime.type = 0, then it’ll use the current time, ie time will be respected (incl restrictions, but also traffic if it’s loaded). When you don’t want to respect time, then you don’t send the date_time object in the request at all. And I agree, the latter will be the vast majority of requests.

File: src/components/datetime-picker.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'react'
```
Comment (chrstnbwnkl) [inline src/components/datetime-picker.jsx:1]: File name should be pascal case

File: src/components/datetime-picker.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import PropTypes from 'prop-types'
+import { Input, Dropdown } from 'semantic-ui-react'
+
+export const DateTimePicker = ({ type, value, onChange }) => {
+  return (
+    <div
+      className="pa2 flex flex-wrap justify-between"
+      style={{
+        marginLeft: 'auto',
```
Comment (chrstnbwnkl) [inline src/components/datetime-picker.jsx:10]: doesn't seem to have any effect

File: src/Controls/Directions/index.jsx [hunk 3]
```diff
@@ -62,8 +68,24 @@ class DirectionsControl extends React.Component {
...
+      date_time.value !== nextProps.date_time.value
+    if (shouldUpdate) {
+      this.props.dispatch(makeRequest())
+    }
+    return shouldUpdate
+  }
+
   render() {
-    const { profile, loading } = this.props
+    const { profile, loading, date_time } = this.props
```
Comment (chrstnbwnkl) [inline src/Controls/Directions/index.jsx:88]: again, please use camelCase

**Review by chrstnbwnkl — CHANGES_REQUESTED** [review]
Thanks @nvcnvn , just a couple of comments!

**Comment by nvcnvn** [comment]
> Thanks @nvcnvn , just a couple of comments!

Finally I found some time to update the PR.

**Review by chrstnbwnkl — CHANGES_REQUESTED** [review]
Sorry for not getting back to this earlier... Looks like you forgot to check in the renamed datepicker component file?

**Comment by nvcnvn** [comment]
> Sorry for not getting back to this earlier... Looks like you forgot to check in the renamed datepicker component file?

ah you are right

**Review by chrstnbwnkl — APPROVED** [review]
...and again sorry for the long wait. LGTM!

**Pull request merged**